lib/file_transfer_agent/encrypt_util.js

```diff
@@ -32,6 +35,61 @@ function EncryptionMetadata(key, iv, matDesc) {
 
 exports.EncryptionMetadata = EncryptionMetadata;
 
+function TempFileGenerator() {
+
+  this.fileSync = function (option = { dir: os.tmpdir(), prefix: '', postfix: '', extension: '' }) {
+    const randomName = crypto.randomUUID();
+    const fileName = `${option.prefix || ''}${randomName}${option.postfix || ''}${'.' + option.extension || ''}`;
+      
+    if (!this.checkDirInTemp(option.dir)) {
+      option.dir = os.tmpdir();
+    }
+
+    const fullpath = path.join( option.dir, fileName);
+    fs.writeFileSync(fullpath, '');
+
+    const fileDescripter = fs.openSync(fullpath);
+    return { name: fullpath, fd: fileDescripter };
+  };
+  
+  this.file = function (option = { dir: os.tmpdir(), prefix: '', postfix: '', extension: '' }, callback) {
+    const randomName = crypto.randomUUID();
+    const fileName = `${option.prefix || ''}${randomName}${option.postfix || ''}${'.' + option.extension || ''}`;
+
+    if (!this.checkDirInTemp(option.dir)) {
+      option.dir = os.tmpdir();
+    }
+
+    const fullpath = path.join( option.dir, fileName);
```

Comment:
The part of creating fullpath is duplicated, what do you think about exctracting?

Reply:
I made another method to get the full path.

Reply:
I think about even more duplicated lines :

Reply:
You are right. I read my code again and realized that the filesync and the file were using the same code until we read the descriptor. As a result, I called the filesync in the file function, and return a callback. Please have a look.